ogre2/src/Ogre2ThermalCamera.cc

```diff
@@ -286,30 +287,36 @@ void Ogre2ThermalCameraMaterialSwitcher::preRenderTargetUpdate(
             {
               ignerr << "Error casting user data: " << e.what() << "\n";
               temp = -1.0;
+              foundTemp = false;
             }
           }
         }
 
-        // only accept positive temperature (in kelvin)
-        if (temp >= 0.0)
+        // if a non-positive temperature was given, clamp it to 0
+        if (foundTemp && temp < 0.0)
```

Comment:
This only clamps the `temp` value when `foundTemp` is true. Is it possible for `foundTemp` to be false and `temp` to be < 0.0? If so, should you handle that case?

Reply:
> Is it possible for foundTemp to be false and temp to be < 0.0?

I don't think so. The only way `foundTemp` is set to true is if `temperature` is defined/attached to a visual, which would mean that `<temperature>` needs to be specified in the SDF file in the `ign-gazebo` context. If `<temperature>` is not specified, then this code should run (background object where object colors are used to define temperatures - since we're using rgb here, it shouldn't be possible to have an rgb value that is < 0): https://github.com/ignitionrobotics/ign-rendering/blob/8e0de0dd3e499c0fc0191a2cd8b794b0aab69980/ogre2/src/Ogre2ThermalCamera.cc#L387-L417